src/Appwrite/Platform/Workers/Builds.php

```diff
@@ -390,6 +390,10 @@ protected function buildDeployment(callable $getFunctionsDevice, Func $queueForF
             $response = null;
             $err = null;
 
+            if ($build->getAttribute('status') === 'cancelled') {
```

Comment:
How are we actually letting open runtimes to stop the runtime build and save compute time?

Reply:
This command stops the runtime and kills the container: https://github.com/appwrite/appwrite/blob/6ebca949c0bd86442a826cd272b7ad197b3068a2/app/controllers/api/functions.php#L1516

If the build is cancelled before the container is created, we're preventing the creation of the container here: https://github.com/appwrite/appwrite/blob/6ebca949c0bd86442a826cd272b7ad197b3068a2/src/Appwrite/Platform/Workers/Builds.php#L393

Let me know if you have any questions.